flink/src/main/java/org/apache/iceberg/flink/IcebergTableSink.java

```diff
@@ -63,6 +72,7 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
         .tableLoader(tableLoader)
         .tableSchema(tableSchema)
         .equalityFieldColumns(equalityColumns)
+        .upsert(PropertyUtil.propertyAsBoolean(tableProperties, UPSERT_MODE_ENABLED, UPSERT_MODE_ENABLED_DEFAULT))
```

Comment:
do we need to pass in the table properties for the upsert flag? `FlinkSink.Builder.build()` method internally open the table from TableLoader (if not passed in). Can we extract the prop inside the `build()` method.

Also we probably can add a validation that `upsert` and `overwrite` aren't enabled at the same time?

Reply:
Sounds good to me! I will extract this prop in `build()` and add a validation for `overwrite` when `upsert` is enable. :)